prime-router/build.gradle.kts

```diff
@@ -0,0 +1,312 @@
+        val formattedDate = LocalDateTime.now().format(dateFormatter)
+        val tokens = mapOf("version" to version, "timestamp" to formattedDate)
+        filter(ReplaceTokens::class, mapOf("beginToken" to "@", "endToken" to "@", "tokens" to tokens))
+    }
+}
+
+tasks.jar {
+    manifest {
+        /* We put the CLI main class in the manifest at this step as a convenience to allow this jar to be
+        run by the ./prime script. It will be overwritten by the Azure host or the CLI fat jar package. */
+        attributes("Main-Class" to primeMainClass)
+    }
+}
+
+tasks.register<JavaExec>("primeCLI") {
+    group = rootProject.description ?: ""
+    description = "Run the Prime CLI tool.  Specify arguments with --args='<args>'"
+    main = primeMainClass
+    classpath = sourceSets["main"].runtimeClasspath
+    // Default arguments is to display the help
+    args = listOf("-h")
+    environment = mapOf("POSTGRES_URL" to dbUrl, "POSTGRES_USER" to dbUser, "POSTGRES_PASSWORD" to dbPassword)
+    doFirst() {
+        println("primeCLI Gradle task usage: gradle primeCLI --args='<args>'")
+    }
+}
+
+tasks.register<JavaExec>("testEnd2End") {
+    group = rootProject.description ?: ""
+    description = "Run the end to end tests.  Requires running a Docker instance"
+    main = primeMainClass
+    classpath = sourceSets["main"].runtimeClasspath
+    args = listOf("test", "--run", "end2end")
+    environment = mapOf("POSTGRES_URL" to dbUrl, "POSTGRES_USER" to dbUser, "POSTGRES_PASSWORD" to dbPassword)
+}
+
+azurefunctions {
+    appName = azureAppName
+    setAppSettings(closureOf<MutableMap<String, String>> {
+        this["WEBSITE_RUN_FROM_PACKAGE"] = "1"
+        this["FUNCTIONS_EXTENSION_VERSION"] = "3"
+        this["FUNCTIONS_WORKER_RUNTIME"] = "java"
+    })
+}
+
+tasks.azureFunctionsPackage {
+    dependsOn("test")
+}
+
+val azureResourcesTmpDir = File(rootProject.buildDir.path, "${azureFunctionsDir}-resources/${azureAppName}")
+val azureResourcesFinalDir = File(rootProject.buildDir.path, "${azureFunctionsDir}/${azureAppName}")
+tasks.register<Copy>("gatherAzureResources") {
+    from("./")
+    into(azureResourcesTmpDir)
+    include("metadata/**/*.yml")
+    include("metadata/**/*.schema")
+    include("metadata/**/*.valuesets")
+    include("metadata/**/*.csv")
+    include("settings/**/*.yml")
+    include("assets/**/*__inline.html")
+}
+
+tasks.register("copyAzureResources") {
+    dependsOn("gatherAzureResources")
+    doLast() {
+        // We need to use a regular copy, so Gradle does not delete the existing folder
+        org.apache.commons.io.FileUtils.copyDirectory(azureResourcesTmpDir, azureResourcesFinalDir)
+    }
+}
+
+val azureScriptsTmpDir = File(rootProject.buildDir.path, "${azureFunctionsDir}-scripts/${azureAppName}")
+val azureScriptsFinalDir = rootProject.buildDir
+tasks.register<Copy>("gatherAzureScripts") {
+    from("./")
+    into(azureScriptsTmpDir)
+    include("prime")
+    include("start_func.sh")
+}
```

Comment:
If you notice, the POM does a chmod as well as copies these files. 
```
            <plugin>
                <artifactId>maven-antrun-plugin</artifactId>
                <version>3.0.0</version>
                <executions>
                    <execution>
                        <phase>package</phase>
                        <configuration>
                            <target>
                                <copy file = "prime" tofile = "target/prime"/>
                                <chmod file = "target/prime" perm = "+x"/>
                                <copy file = "start_func.sh" tofile = "target/start_func.sh"/>
                                <chmod file = "target/start_func.sh" perm = "+x"/>
                            </target>
                        </configuration>
                        <goals>
                            <goal>run</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```
Marking these files execute is needed to make `docker-compose up` work.

Reply:
Done.  I can see my changes took effect, but I am unable to fully test in Windows as I never has an issue.  Please pull and retest doing a full build:
`./gradlew clean package`

Reply:
@RickHawesUSDS Arg!  I am changing my approach.  Could you pull and try again?

Reply:
I see the problem. Change the code to set the file in `azureScriptsFinalDir`... 
```
        File(azureScriptsFinalDir.path, primeScriptName).setExecutable(true);
        File(azureScriptsFinalDir.path, startFuncScriptName).setExecutable(true);
```

Reply:
With this change, I can run the functions locally on my mac.

Reply:
BTW: The pulled the latest and that doesn't work. I think your latest changes didn't fix the problem, but your previous approach did work with my change.

Reply:
Thank you.  I really wanted to find a way that I could truly verify in Windows, but seems that is not going to be the case.  I'll use the approach you took.  :(

Reply:
Current branch is working for me.